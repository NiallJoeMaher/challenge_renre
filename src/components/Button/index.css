.c-Button {
  font-family: 'Fredericka the Great', cursive;
  font-size: 16px;
  border: none;
  cursor: pointer;
  color: #393f4d;
  flex: 1 ;
  height: 40px;
  background-color: #d4d4dc;
  transition: all .3s ease-in-out;
}

.c-Button--active {
  font-family: 'Fredericka the Great', cursive;
  font-size: 16px;
  border: none;
  cursor: pointer;
  color: #393f4d;
  flex: 1 ;
  height: 40px;
  background-color: #d4d4dc;
  transition: all .3s ease-in-out;
  box-shadow: inset 0px 0px 15px 1px rgba(0,0,0,0.45);
}

.c-Button:hover {
  background: #feda6a;
}

.c-Button--active:hover {
  background: #feda6a;
}
