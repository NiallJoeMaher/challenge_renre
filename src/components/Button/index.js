import React, { Component } from 'react';
import './index.css';

import PropTypes from 'prop-types';

class Button extends Component {
  static propTypes = {
    text: PropTypes.string.isRequired,
    onClick: PropTypes.func.isRequired,
    active: PropTypes.bool
  };

  static defaultProps = {
    active: false
  }

  render() {
    const { onClick, text, active } = this.props;
    const className = `c-Button${active ? '--active' : ''}`;
    return (
      <button 
        onClick={onClick} 
        key={text} 
        className={className}
      >
      {text}
      </button>
    )
  }
}

export default Button;
