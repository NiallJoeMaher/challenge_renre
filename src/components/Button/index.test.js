import React from 'react';
import Button from './';
import { shallow } from 'enzyme';

it('renders as expected', () => {
  const component = shallow(<Button text='Hello World' onClick={() => {}} />);
  expect(component).toMatchSnapshot();
});

it('Test click event', () => {
  const mockCallBack = jest.fn();
  const button = shallow(<Button onClick={mockCallBack} text='Ok!' />);
  button.find('button').simulate('click');
  expect(mockCallBack.mock.calls.length).toEqual(1);
});
