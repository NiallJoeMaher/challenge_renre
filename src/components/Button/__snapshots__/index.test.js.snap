// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<button
  className="c-Button"
  key="Hello World"
  onClick={[Function]}
>
  Hello World
</button>
`;
